### ISSUE TITLE
Choose CSP to be used for Optimizer

### ISSUE THREAD

**Issue by simanadler** [fybrik/fybrik#1132]
Evaluate which tools should be used for constraint solving in the optimizer.  There are two aspects:
1) The format/language in which the CSP model is written
2) The back end tool used to evaluate 

Ideally we would like the model authoring to support multiple CSP evaluators, so that we don't lock our users to a particular CSP engine.  Ex: minizinc

See discussions in:
* Last slide of https://drive.google.com/file/d/1wmU07hwZpTY3FbJaziZhYZEw21si0QHw/view?usp=sharing
* https://docs.google.com/document/d/171TWD3FPEV-ZNoO_f8vQQoj-EV-wy6kuq6IyQn6QPPY/edit?usp=sharing

**Comment by zivnevo** [comment]
Following a discussion on Feb 3, it was decided to go for [FlatZinc](https://www.minizinc.org/doc-2.5.5/en/fzn-spec.html#specification-of-flatzinc)  as the CSP language (or MiniZinc if this is too hard) and for [Google's OR tools](https://github.com/google/or-tools) as the first solver to use.
See the [presentation](https://docs.google.com/presentation/d/1A8kSsd044Jr0cGQgO0VCNaZfrzo349Q1/edit?usp=sharing&ouid=109239644760744081579&rtpof=true&sd=true) and [excel sheet](https://docs.google.com/spreadsheets/d/1jTe8T38qfbv4yRlq1-QcrHVZASJkJFBB/edit?usp=sharing&ouid=109239644760744081579&rtpof=true&sd=true) from the discussion.

**Comment by roytman** [comment]
@zivnevo @simanadler based on the comments and discussions looks like the issue can be closed.

**Comment by simanadler** [comment]
@zivnevo Please add what was chosen and why, and then we can close this issue

**Comment by zivnevo** [comment]
After implementing most of the required functionality, it seems that OR tools suits our needs.
So far, I found two non-blocking limitations of Google's OR tools:
1. No support for arrays of sets of integers (our implementation can easily work around it)
2. No support for floating-point variables/constants (means we have to convert them to integers and set the required precision)

Closing the issue for now.
